src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1804,14 +1820,31 @@ private void visitEnumType(String symbolName, String qualifiedName, EnumType typ
         return;
       }
 
-      JSType primitiveType = type.getElementsType().getPrimitiveType();
-      if (maybeStringOrNumericEnum(primitiveType, unqualifiedName, node)) {
+      // The current node points to the enum type declaration, this means that the next node will
+      // be the OBJECTLIT containing all enum key and value pairs. However, globally declared
+      // enums that are indirectly provided will instead be pointing to the parent of the
+      // OBJECTLIT parent.
+      Node objectOfAllMembers = node.getNext() != null ? node.getNext() : node.getFirstChild();
+      Stream<Node> elementStream = Streams.stream(objectOfAllMembers.children());
+      Map<String, Node> elements =
+          elementStream.collect(Collectors.toMap(Node::getString, Node::getFirstChild));
+
+      JSType primitiveType = type.getEnumeratedTypeOfEnumObject();
+      if (maybeStringOrNumericEnum(primitiveType, unqualifiedName, objectOfAllMembers, elements)) {
         return;
       }
 
       // Since this is an enum of type different from string literals or numbers, we cannot emit it
-      // as TS enum. Instead we emit it as a var/type alias pair.
-      visitTypeAlias(type.getElementsType().getPrimitiveType(), unqualifiedName, true);
+      // as TS enum. Instead we emit it as a var/type alias pair. We also inline all literal types
```

Comment:
'Specifically for string enums, we also inline ...'

Reply:
Reworded a bit.